src/expr/impl/src/scalar/random.rs

```diff
@@ -0,0 +1,30 @@
+// Copyright 2025 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use rand::Rng;
+use risingwave_common::types::F64;
+use risingwave_expr::function;
+
+/// Generates a random float between 0 and 1.
+///
+/// ```slt
+/// query I
+/// SELECT random();
+/// ----
+/// 0.123456789
+/// ```
+#[function("random() -> float8", volatile)]
+fn random() -> F64 {
+    rand::rng().random::<F64>()
```

Comment:
Use random_range instead?

https://docs.rs/rand/latest/rand/fn.random_range.html

I don't think `random` will have its range constrainted to [0, 1].

Reply:
sure updated
i though random() was a bit faster and we could change the range to [0, 1)  but either ways works

Reply:
i did some research:

random() for floats is just:
Generate a random integer (typically 64 bits).
Convert it into a float in [0, 1) by scaling.
That’s a couple of instructions, with almost no branching.

random_range / Uniform has extra steps:
Store and validate the range bounds.
Apply a scaling transform to map the generated number into the given range.
Sometimes do an extra check for inclusivity or deal with open/closed boundaries.

but honestly the difference is trival